### ISSUE TITLE
The built library does not include assets correctly in nx project？

### ISSUE THREAD

**Issue by luojinghui** [nrwl/nx#27351]
### Current Behavior

I used `nx` + `nx/js` + `vite` to build a js library and configured the assets resources, but the build package did not include the resources in the assets, such as the `README.md` file.

<img width="1062" alt="image" src="https://github.com/user-attachments/assets/afb7db2f-9785-451a-8836-56eef0f9a9db">

### Expected Behavior

When using `nx build` or `nx release`, resources with assets can be correctly output, such as `README.md` files, which is very important for npm packages

### GitHub Repo

https://github.com/luojinghui/nx-project

### Steps to Reproduce

1. clone github repo
2. install dep: `pnpm install`
3. build packages lib: `nx run rtc-sdk:build`
4. check the `dist` directory to see if the `README.md` file exists?

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node   : 20.14.0
OS     : darwin-x64
pnpm   : 8.15.9

nx (global)        : 19.5.7
nx                 : 19.5.3
@nx/js             : 19.5.4
@nx/jest           : 19.5.3
@nx/eslint         : 19.5.3
@nx/workspace      : 19.5.3
@nx/eslint-plugin  : 19.5.3
@nx/react          : 19.5.3
@nx/vite           : 19.5.4
@nx/vue            : 19.5.4
@nx/web            : 19.5.4
@nx/webpack        : 19.5.3
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/webpack/plugin
@nx/eslint/plugin
@nx/jest/plugin
@nx/vite/plugin
---------------------------------------
The following packages should match the installed version of nx
  - @nx/js@19.5.4
  - @nx/vite@19.5.4
  - @nx/vue@19.5.4
  - @nx/web@19.5.4

To fix this, run `nx migrate nx@19.5.4`
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information
if use `@nx/js:tsc` executor to build, it's no problem!

_No response_

**Comment by luojinghui** [comment]
hi, Is there any progress on the issue? @jaysoo

**Comment by jaysoo** [comment]
Hi @luojinghui, the `@nx/vite:build` doesn't support `assets`. Our philosophy is to rely on the underlying tool as much as possible, so in this case I would configure assets in `vite.config.ts` instead. Maybe something like this would help: https://www.npmjs.com/package/vite-plugin-static-copy

I think we can add a plugin for this and configure `vite.config.ts` automatically for JS libs. That would be a new feature.

**Comment by jaysoo** [comment]
I opened a PR for this. You can configure the plugin like:

```js
// Copies all markdown files to dist
nxCopyAssetsPlugin(['*.md'])
```

In the meantime I think `vite-plugin-static-copy` will still work.

**Comment by luojinghui** [comment]
@jaysoo 
Thank you. Actually, I tried to solve the problem with the two plugins `rollup-plugin-copy` and `vite-plugin-static-copy` before. Unfortunately, they both had problems. It seemed that the problem was caused by the timing difference between nx build and plugin execution. In the end, the specified `*.md` resource was not successfully copied to the destination.

So later I tried to execute a custom nodejs script after the nx build was completed to solve the problem, but it was not so elegant.

Thank you for providing another idea. I tried to upgrade the @nx/vite version to: @nx/vite@19.6.3, but did not find the corresponding plugin: @nx/vite/plugins/nx-copy-assets.plugin. Which version do I need to install?
